### PR TITLE
Nicer names for cacherecord

### PR DESCRIPTION
**Pull request by AdamBJohnsonx**
This started out smaller, put updating the common submodule caused a truly strange error.  In the course of
debugging, I've made the failure messages coming from some of our tests much more verbose, so that I can find
the errors without needing to use a debugger.

### CODE REVIEW

**Review by iambmelt — APPROVED** [review]
LGTM

File: msal/build.gradle [hunk 1]
```diff
@@ -143,7 +142,9 @@ android {
                 // example: ./gradlew testLocalDebugUnitTest -Plabtest
                 exclude 'com/microsoft/identity/client/e2e/tests/network'
             }
-
+            testLogging {
+                exceptionFormat = "full"
```
Comment (iambmelt) [inline msal/build.gradle:146]: 👍

File: gradle/versions.gradle [hunk 2]
```diff
@@ -36,7 +36,7 @@ ext {
     mockitoAndroidVersion = "3.6.28"
     multidexVersion = "2.0.1"
     nimbusVersion = "8.2"
-    powerMockVersion = "1.6.6"
+    powerMockVersion = "2.0.9"
```
Comment (shahzaibj) [inline gradle/versions.gradle:39]: Can we update in other places such as android-complete as well?

**Review by iambmelt — APPROVED** [review]

**Review by rpdome — APPROVED** [review]

File: msal/src/test/java/com/microsoft/identity/client/e2e/tests/mocked/SingleAccountAcquireTokenMockTest.java [hunk 3]
```diff
@@ -66,7 +66,8 @@ public void onAccountChanged(@Nullable IAccount priorAccount, @Nullable IAccount
 
             @Override
             public void onError(@NonNull MsalException exception) {
-                fail("No current account found.");
+                fail("Interpreting an exception as no accounts found.  Unexpected exception: " + exception.getMessage() +
```
Comment (shahzaibj) [inline msal/src/test/java/com/microsoft/identity/client/e2e/tests/mocked/SingleAccountAcquireTokenMockTest.java:69]: could also do `throw new AssertionError(exception)` and then we won't have to log the exception message and stack trace ourselves.

File: gradle/versions.gradle (hunk 2 above)
Comment (AdamBJohnsonx) [reply gradle/versions.gradle:39]: I think android-complete is updated.

Comment (AdamBJohnsonx) [reply gradle/versions.gradle:39]: nope/

**Review by shahzaibj — APPROVED** [review]
Just some feedback around changing `fail` and assert statements to `throw new AssertionError(exception)`

**Pull request merged**